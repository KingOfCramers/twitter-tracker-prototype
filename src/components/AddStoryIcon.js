import React from 'react';
import { connect } from "react-redux";
import { startAddStory, startRemoveStory } from "../actions/stories";

export const AddStoryIcon = (props) => (
  <div>
    <button className="button" onClick={props.addStory}>Add Story</button>
  </div>
);

const mapDispatchToProps = (dispatch, props) => ({
  addStory: () => dispatch(startAddStory({ story: "This is new", deadline: 19039 })),
})

export default connect(undefined, mapDispatchToProps)(AddStoryIcon);