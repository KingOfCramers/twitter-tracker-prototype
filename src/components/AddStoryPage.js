import React from 'react';

const AddStoryPage = () => (
  <div>
    Add Story Page
  </div>
);

export default AddStoryPage;