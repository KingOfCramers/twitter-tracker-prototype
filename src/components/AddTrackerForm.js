import React from "react";

class AddTrackerForm extends React.Component {
  render(props){
    return (
      <div>
        {this.props.type}
      </div>
    );
  }
}

export default AddTrackerForm;