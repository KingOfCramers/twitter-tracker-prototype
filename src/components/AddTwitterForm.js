import React from "react";
import { startAddHandle } from "../actions/twitter";
import { connect } from 'react-redux';

export class AddTwitterForm extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      handle: "",
      error: ""
    };
  };

  onHandleChange = (e) => {
    const handle = e.target.value;
    this.setState(() => ({ handle }));
  };

  onSubmit = (e) => {
    e.preventDefault();
    if(!this.state.handle){
         this.setState(() => ({ error: "Please add a handle."}))
    } else {
      this.props.startAddHandle({ handle: this.state.handle, story_id: this.props.story_id })
      .then(() => this.setState({ handle: "", error: "" }))
    }
  };

  render(){
    return (
      <div>
        {this.state.error && <p className="form__error">{this.state.error}</p>}
        <form>
          <input
            type="text"
            placeholder="handle..."
            autoFocus
            value={this.state.handle}
            onChange={this.onHandleChange}
            className="text-input"
            required
          />
          <button onClick={this.onSubmit} className="button">Add Handle</button>
        </form>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  startAddHandle: (handle) => dispatch(startAddHandle(handle))
});

export default connect(undefined, mapDispatchToProps)(AddTwitterForm);