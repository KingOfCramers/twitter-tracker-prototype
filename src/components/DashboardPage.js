import React from 'react';
import { Link } from 'react-router-dom';

import StoryList from "./StoryList"; // Dashboard page fetches the data about all the stories someone has in the database and diplays the correct number of stories accordingly.

export class DashboardPage extends React.Component {

  onButtonClick = () => {
    this.props.history.push("/story/create");
  }

  render(){
    return (
      <div className="content-container">
         <StoryList />
         <button className="button" onClick={this.onButtonClick}>Add Story</button>
      </div>
    );
  }
}


export default DashboardPage;