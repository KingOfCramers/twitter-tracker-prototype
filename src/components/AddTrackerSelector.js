import React from "react";
import AddTwitterForm from "./AddTwitterForm";
import TrackerDisplay from "./TrackerDisplay";

export class AddTrackerSelector extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      type: 'Twitter'
    };
    this.handleChange = this.handleChange.bind(this);
    this.pickForm = this.pickForm.bind(this);
  };

  handleChange(event) {
    this.setState({ type: event.target.value });
  };

  pickForm(){
    switch(this.state.type) {
      case "Twitter" :
        return <AddTwitterForm story_id={this.props.story_id}/>
      default :
        return <p>Nothing!</p>
    };
  };

  render() {
    return (
      <div>
      <form>
        <label>
          <select type={this.state.type} onChange={this.handleChange}>
            <option type="Twitter">Twitter</option>
            <option type="Legislation">Legislation</option>
            <option type="Lobbying">Lobbying</option>
            <option type="Court Cases">Court Cases</option>
            <option type="Federal Register">Federal Register</option>
            <option type="Campaigns">Campaigns</option>
          </select>
        </label>
      </form>
      {this.pickForm()}
      <TrackerDisplay story_id={this.props.story_id} type={this.state.type.toLowerCase()}/>
      </div>
    );
  }
}

export default AddTrackerSelector;