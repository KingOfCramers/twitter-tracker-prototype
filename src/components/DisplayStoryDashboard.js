import React from 'react';
import { connect } from "react-redux";
import moment from "moment";
import AddTrackerSelector from "./AddTrackerSelector";
import { startRemoveStory } from "../actions/stories";
import { Link } from "react-router-dom";

export class EditStoryDashboard extends React.Component {

  onBack = () => {
    this.props.history.push("/");
  };

  onEdit = () => {
    this.props.history.push(`/story/${this.props.story.id}/edit`);
  };

  render(props){
    return (
    <div className="content-container">
      <button onClick={this.onBack} className="button--secondary">Home</button>
      <button onClick={this.onEdit} className="button">Edit</button>
      <div className='story-info'>
      <h2>{this.props.story.story}</h2>
        <p>{moment(this.props.story.dueDate).format("LL")}</p>
        <p>{this.props.story.description}</p>
      </div>
    </div>
    );
  }
}

const mapStateToProps = (state, props) => ({ // we have access to the props as the second argument here.
    story: state.stories.find((story) => story.id === props.match.params.id) // the props.match.params comes from the Router (it's part of the URL)
});

export default connect(mapStateToProps)(EditStoryDashboard);