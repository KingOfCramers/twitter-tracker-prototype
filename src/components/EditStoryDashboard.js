import React from 'react';
import { connect } from "react-redux";
import moment from "moment";
import AddTrackerSelector from "./AddTrackerSelector";
import { startRemoveStory } from "../actions/stories";
import { Link } from "react-router-dom";

export class EditStoryDashboard extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      display: "twitter"
    }
  };

  onView = () => {
    this.props.history.push(`/story/${this.props.story.id}`);
  };

  onSettings = () => {
    this.props.history.push(`/story/${this.props.story.id}/edit/settings`);
  };

  render(props){
    return (
     <div className="content-container">
     <div className='story-info'>
       <h2>{this.props.story.story}</h2>
       <p>{this.props.story.description}</p>
       <p>{moment(this.props.story.dueDate).format("LL")}</p>
     </div>
      <button className="button--secondary" onClick={this.onView}>Back</button>
      <button className="button" onClick={this.onSettings}>Settings</button>
      <div>
        <AddTrackerSelector story_id={this.props.story.id}/>
      </div>
    </div>
    );
  }
}

const mapStateToProps = (state, props) => ({ // we have access to the props as the second argument here.
    story: state.stories.find((story) => story.id === props.match.params.id) // the props.match.params comes from the Router (it's part of the URL)
});

export default connect(mapStateToProps)(EditStoryDashboard);