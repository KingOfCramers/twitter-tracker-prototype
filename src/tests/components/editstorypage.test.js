import React from "react";
import { EditStoryDashboard } from "../../components/EditStoryDashboard";
import { shallow } from "enzyme";

test("Should render story page correctly", () => {
  const props = {
    story: "STory title",
    dueDate: 123092408,
    description: "soidf0qiof"
  }
  const wrapper = shallow(<EditStoryDashboard {...props}/>);
  expect(wrapper).toMatchSnapshot();
});