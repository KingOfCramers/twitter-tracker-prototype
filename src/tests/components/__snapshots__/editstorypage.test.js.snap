// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render story page correctly 1`] = `
<div
  className="content-container"
>
  <div
    className="story-info"
  >
    <h2 />
    <p />
    <p>
      December 31, 1969
    </p>
  </div>
  <button
    className="button--secondary"
    onClick={[Function]}
  >
    Back
  </button>
  <button
    className="button"
    onClick={[Function]}
  >
    Settings
  </button>
  <div>
    <AddTrackerSelector />
  </div>
</div>
`;
